app/src/main/java/com/arctouch/codechallenge/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.arctouch.codechallenge.ui.home
+
+import android.annotation.SuppressLint
+import android.arch.lifecycle.ViewModel
+import android.util.Log
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.util.Logger
+import io.reactivex.Observable
+
+class HomeViewModel : ViewModel() {
+    private var TAG = Logger.tag
+
+    var cacheMovies = mutableListOf<Movie>()
+
+    var isLoading: Boolean = false
+        private set
+    var currentPage: Long = -1
+        private set
+
+    /**
+     * Method to the request all movies in the page parameter.
+     * @param page [Long]
+     */
+    @SuppressLint("CheckResult")
+    fun requestMovies(page: Long): Observable<Movie>? {
```

Comment:
What would be your strategy to unit test this method? Can you foresee any challenges with your current implementation? Would you consider mocking the API object (`TmdbApi`)?

Reply:
In this method I would test if the Observable is handling the changes, I do not see need to mocking data in this case.